staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -1406,14 +1419,66 @@ func CreateThreeWayMergePatch(original, modified, current []byte, dataStruct int
 	return json.Marshal(patchMap)
 }
 
-func ItemAddedToModifiedSlice(original, modified string) bool { return original > modified }
+func ItemAddedToModifiedSlice(compareResult mergeKeysCompareResultType) bool {
+	return compareResult == mergeKeysGreater
+}
+
+func ItemRemovedFromModifiedSlice(compareResult mergeKeysCompareResultType) bool {
+	return compareResult == mergeKeysLess
+}
+
+func ItemMatchesOriginalAndModifiedSlice(compareResult mergeKeysCompareResultType) bool {
+	return compareResult == mergeKeysEqual
+}
+
+type mergeKeysCompareResultType string
+
+const (
+	mergeKeysLess     mergeKeysCompareResultType = "mergeKeysLess"
+	mergeKeysGreater  mergeKeysCompareResultType = "mergeKeysGreater"
+	mergeKeysEqual    mergeKeysCompareResultType = "mergeKeysEqual"
+	mergeKeysNotFound mergeKeysCompareResultType = "mergeKeysNotFound"
+)
 
-func ItemRemovedFromModifiedSlice(original, modified string) bool { return original < modified }
+func compareByMergeKeys(left, right map[string]interface{}, mergeKeys []string) mergeKeysCompareResultType {
+	var leftValue, rightValue interface{}
+	var leftString, rightString string
+	var foundLeft, foundRight bool
+	for _, mergeKey := range mergeKeys {
+		if leftValue, foundLeft = left[mergeKey]; foundLeft {
+			leftString = fmt.Sprintf("%v", leftValue)
+		} else {
+			return mergeKeysNotFound
+		}
+		if rightValue, foundRight = right[mergeKey]; foundRight {
+			rightString = fmt.Sprintf("%v", rightValue)
+		} else {
+			return mergeKeysNotFound
+		}
+		if leftString != rightString {
```

Comment:
Yes.
We compare by strings, because comparing interfaces can only returns equal or not equal. But when comparing strings we have `<`, `>` and equal. I'm trying to make it consistent.
Do you have any suggestions how to compare interfaces?